airflow/kubernetes/pod_generator.py

```diff
@@ -287,7 +287,8 @@ def reconcile_specs(
             client_spec.containers = PodGenerator.reconcile_containers(
                 base_spec.containers, client_spec.containers
             )
-            merged_spec = extend_object_field(base_spec, client_spec, 'volumes')
+            merged_spec = extend_object_field(base_spec, client_spec, 'init_containers')
+            merged_spec = extend_object_field(base_spec, merged_spec, 'volumes')
```

Comment:
can this be renamed to merged_volume_spec and merged_init_container_spec for clarity.

Reply:
Hello, thanks for the suggestion, but I think doing that will be less clear because it will be something like this:
```
            merged_init_container_spec = extend_object_field(
                base_spec,
                client_spec,
                'init_containers')
            merged_volume_spec = extend_object_field(
                base_spec,
                merged_init_container_spec,
                'volumes')
            return merge_objects(base_spec, merged_volume_spec)
```

My idea was that you have a `base_spec`, a `client_spec` and you get a `merged_spec` extending both volumes and init_containers. Finally, you merge the object, by overwriting the `base_spec` with the `merged_spec`.

But please, let me know if you see it differently or you have a better approach.

Reply:
The concern is more on line 291 where merged_spec is passed to the function and the return value is assigned back to merged_spec, someone making change in the future might not notice it

Reply:
I'm looking at `extend_object_field`'s use in `reconcile_metadata`, maybe it'd be more clear like this:

```
            extend_object_field(base_spec, client_spec, 'init_containers')
            extend_object_field(base_spec, client_spec, 'volumes')
            return merge_objects(base_spec, client_spec)
```

Reply:
I'm not an expert in python, but looking into the function `extend_object_field` it seems that is returning a copy and not changing any of the arguments. So I don't know exactly how reconcile_metatadata is working but if I change the code to this:
```
            extend_object_field(base_spec, client_spec, 'init_containers')
            extend_object_field(base_spec, client_spec, 'volumes')
            return merge_objects(base_spec, client_spec)
```
The test fails:
`tests/kubernetes/test_pod_generator.py::TestPodGenerator::test_reconcile_specs_init_containers FAILED                                      [ 96%]`


I've thinking another less confusing code and it could be this:
```
            extended_spec = copy.deepcopy(client_spec)
            extended_spec = extend_object_field(base_spec, extended_spec, 'init_containers')
            extended_spec = extend_object_field(base_spec, extended_spec, 'volumes')
            return merge_objects(base_spec, extended_spec)
```

What do you think?

Thanks.

Reply:
I'm good with either `extended_spec` or `merged_spec`. The variable is only in play for 3 lines anyways 🤷‍♂️.
(I don't think we should deepcopy it explicitly)

Reply:
So, then we keep the current code ?
Do you agree @subkanthi ?